### ISSUE TITLE
Integrating with sbt-dependency-check plugin

### ISSUE THREAD

**Issue by maoo** [ckipp01/mill-github-dependency-graph#44]
Hi there! I'm playing with this plugin in order to validate some of [our FINOS](https://landscape.finos.org/) incoming projects, which builds with Mill. Part of our process is to ensure that none of direct or transitive dependencies have HIGH or CRITICAL CVEs, and seems like this is the tool for us; however, since Dependabot doesn't work on transitive dependencies, we normally use https://github.com/albuch/sbt-dependency-check to scan the dependency tree (you can read more on https://github.com/maoo/security-scanning#scala .

I was wondering if there is a way to integrate the mill-github-dependency-grapth with the sbt-dependency-check plugin, and skip the whole part that submits data to GitHub.

I'm not sure if this is a use case that is at all relevant for you, feel free to close this issue if it's not, but even in that case, I thought you could give me some useful pointers, and even some code I could reuse in order to achieve my goal.

Thanks in advance, and congrats for contributing this great tool!

**Comment by ckipp01** [comment]
Hey @maoo, thanks for the question. A couple things:

Firstly, you might have better luck asking about this in https://github.com/scalacenter/sbt-dependency-submission, which is the sbt equivalent for what the plugin is doing if your builds are with sbt. I'm not sure they are if you're using this, but then also saying you're using `sbt-dependency-check`.

> since Dependabot doesn't work on transitive dependencies

So actually with both this mill plugin and the sbt one, your transitive dependencies actually _are_ submitted. So dependabot can warn you about them.

> I was wondering if there is a way to integrate the mill-github-dependency-grapth with the sbt-dependency-check plugin, and skip the whole part that submits data to GitHub.

Probably not as the whole point of this plugin _is_ to submit it to GitHub in order to get dependabot alerts. Having a mill equivalent to `sbt-dependency-check` is a bit out of scope for this project.

Again, thanks for the questions!

**Comment by maoo** [comment]
Thanks for your quick reply @ckipp01 ! My idea is to actually build a `mill-dependency-check` plugin, starting from your code (`Graph.scala`, `ModuleTrees.scala` and `Resolver.scala` would actually be unchanged), adapt `Writers.scala` and add some logic from [DependencyCheckPlugin.scala](https://github.com/albuch/sbt-dependency-check/blob/main/src/main/scala/net/vonbuchholtz/sbt/dependencycheck/DependencyCheckPlugin.scala).

Before starting to code, I'll gather more validations, as I don't want to risk to reinvent the wheel; unfortunately, I've checked the results we get in the GitHub dependency tree, and it doesn't contain all transitive deps; as soon as the repo is public, I can share more info with you, if you're interested.

Thanks so much for your suggestions!

**Comment by ckipp01** [comment]
> Thanks for your quick reply @ckipp01 ! My idea is to actually build a mill-dependency-check plugin, starting from your code (Graph.scala, ModuleTrees.scala and Resolver.scala would actually be unchanged), adapt Writers.scala and add some logic from [DependencyCheckPlugin.scala](https://github.com/albuch/sbt-dependency-check/blob/main/src/main/scala/net/vonbuchholtz/sbt/dependencycheck/DependencyCheckPlugin.scala).

Ah, cool. That'd be really useful!

>  I've checked the results we get in the GitHub dependency tree, and it doesn't contain all transitive deps

Ah really? It definitely should as we resolve everything, so the results should be everything after reconciliation. When the repo is public, for sure report back as I'd like to dig into what is missing. Also make sure to check out the [generate task]( I've checked the results we get in the GitHub dependency tree, and it doesn't contain all transitive deps) at the bottom with the latest version and you'll actually be able to see the full manifest to get a better idea of what might be missing.
